Improve galaxy handling of tgz roles

willthames opened the pull request:
##### ISSUE TYPE
- Bugfix Pull Request
##### ANSIBLE VERSION

```
ansible 2.1.0 (devel 2163a24a12) last updated 2016/04/14 12:28:50 (GMT +1000)
  lib/ansible/modules/core: (detached HEAD 99cd31140d) last updated 2016/04/14 12:29:48 (GMT +1000)
  lib/ansible/modules/extras: (detached HEAD ab2f4c4002) last updated 2016/04/14 12:29:51 (GMT +1000)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

Gerrithub presents tgz downloads that do not have a containing
directory. This causes a stack trace.

As all roles must contain a meta/main.yml, change ansible-galaxy
to use that to determine the enclosing directory (in the case
of multiple meta/main.yml files, use the one with the shortest parent)

Fixes #15413

dmsimard commented on lib/ansible/galaxy/role.py:
```diff
@@ -260,9 +269,9 @@ def install(self):
-                # the tar file here, since the default is 'github_repo-target', and change it
-                # to the specified role's name
+                # we strip off any higher-level directories for all of the files contained within
+                # the tar file here. The default is 'github_repo-target'. Gerrithub, on the other
```
Small nit, Gerrit is the actual software. Gerrithub is a hosted Gerrit instance.

willthames committed:
Improve galaxy handling of tgz roles

Gerrithub presents tgz downloads that do not have a containing
directory. This causes a stack trace.

As all roles must contain a meta/main.yml, change ansible-galaxy
to use that to determine the enclosing directory (in the case
of multiple meta/main.yml files, use the one with the shortest parent)

Fixes #15413
